### PR TITLE
Add support for generating javadoc

### LINKED ISSUE #231: Assemble and deploy javadoc JARs

### PR DESCRIPTION
**Pull request by MihaiBojin**
Please replace every line in curly brackets ( { like this } ) with an appropriate description, and remove this line.

## What is the goal of this PR?

Adds support for assembling Javadoc JARs and deploying them to Maven Central.

Closes #231

## What are the changes implemented in this PR?

- identify the referenced target's `srcs` files
- create a javadoc target which parses all `srcs` and generated docs
- run it as part of `assemble_maven`
- also, allow it to be run individually (e.g., if a developer wants to generate javadocs outside of the assemble target)
- updated the README to list the `srcs` entry

### CODE REVIEW

**Comment by MihaiBojin** [comment]
@vmax can you take a look at this PR when you get the chance, please? Thank you!

**Comment by MihaiBojin** [comment]
Note, I'm using this code in [one of my projects](https://github.com/MihaiBojin/props).

**Comment by MihaiBojin** [comment]
ping @vmax

**Comment by MihaiBojin** [comment]
Hi, @adammitchelldev can you help review this PR, please?

File: maven/templates/rules.bzl [hunk 1]
```diff
@@ -423,6 +440,60 @@ assemble_maven = rule(
...
+
+    # extract paths to use in the javadoc command
+    src_list = []
+    for src in src_files:
+        src_list += [src.path]
+
+    # https://docs.oracle.com/en/java/javase/11/javadoc/javadoc-command.html#GUID-B0079316-8AA3-475B-8276-6A4095B5186A
+    cmd = [
+        "mkdir -p %s" % maven_coordinates.artifact_id,
+        "javadoc -d %s %s" % (maven_coordinates.artifact_id, " ".join(src_list)),
```
Comment (adammitchelldev) [inline maven/templates/rules.bzl:468]: This does not look like the correct way to achieve javadoc generation, considering that it does not use the bazel java toolchain and goes for the system one instead, which may break if the bazel toolchain is a different version. Have a look at this: https://github.com/GerritCodeReview/bazlets/commit/13330934a0e577d60719551ffb746f64b0bb9637 here they get the definite location of the java toolchain being used and use the corresponding javadoc tool directly.

Comment (MihaiBojin) [reply maven/templates/rules.bzl]: Thanks for the review @adammitchelldev .
I updated the PR to call the javadoc provided by the bazel java toolchain.

**Pull request closed without merging**